retrofit-converters/logansquare/src/main/java/retrofit/LoganSquareConverterFactory.java

```diff
@@ -0,0 +1,21 @@
+package retrofit;
+
+import com.bluelinelabs.logansquare.JsonMapper;
+import com.bluelinelabs.logansquare.LoganSquare;
+
+import java.lang.reflect.Type;
+
+/**
+ * A {@linkplain Converter.Factory converter} which uses LoganSquare as its conversion backbone.
+ *
+ * @author Marcel Schnelle (aurae)
+ * @see <a>https://github.com/bluelinelabs/LoganSquare</a>
+ */
+public final class LoganSquareConverterFactory implements Converter.Factory {
+
+	@Override public Converter<?> get(Type type) {
+		// Obtain the JsonMapper for the type and serve a Converter
+		JsonMapper<?> mapper = LoganSquare.mapperFor((Class<?>) type);
```

Comment:
This needs to check if the `type` is not a `Class` and throw an exception with a meaningful message. The implicit `ClassCastException` that's thrown here is unsatisfactory.

Also, does this mean I can't use LoganSquare on a `List` of objects? That's really strange...